File: GXpert/GXpert.Web/Modules/QuestionBank/QuestionOption/QuestionOptionColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.QuestionBank.Columns;

[ColumnsScript("QuestionBank.QuestionOption")]
[BasedOnRow(typeof(QuestionOptionRow), CheckNames = true)]
public class QuestionOptionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
   
    [HeaderCssClass("QuestionOption1")]
    public long Id { get; set; }
    [HeaderCssClass("QuestionOption1"), Hidden]
    public string QuestionText { get; set; }
    [HeaderCssClass("QuestionOption1")]
    [EditLink]
    public string OptionText { get; set; }
    [HeaderCssClass("QuestionOption1")]
    public bool IsCorrect { get; set; }
    [HeaderCssClass("QuestionOption1")]
    public float SortOrder { get; set; }
    [HeaderCssClass("QuestionOption1")]
    public float Weightage { get; set; }
  
}

File: GXpert/GXpert.Web/Modules/ServerTypes/QuestionBank/QuestionOptionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { QuestionOptionRow } from "./QuestionOptionRow";

export interface QuestionOptionColumns {
    Id: Column<QuestionOptionRow>;
    QuestionText: Column<QuestionOptionRow>;
    OptionText: Column<QuestionOptionRow>;
    IsCorrect: Column<QuestionOptionRow>;
    SortOrder: Column<QuestionOptionRow>;
    Weightage: Column<QuestionOptionRow>;
}

export class QuestionOptionColumns extends ColumnsBase<QuestionOptionRow> {
    static readonly columnsKey = 'QuestionBank.QuestionOption';
    static readonly Fields = fieldsProxy<QuestionOptionColumns>();
}